pkg/csi/csi_ops_test.go

```diff
@@ -1160,3 +1159,110 @@ func (s *CSITestSuite) TestDeleteSnapshot(c *C) {
 		c.Check(err, tc.errChecker)
 	}
 }
+
+func (s *CSITestSuite) TestWaitForPVCReady(c *C) {
+	ctx := context.Background()
+	const ns = "ns"
+	const pvc = "pvc"
+	boundPVC := s.getPVC(ns, pvc, v1.ClaimBound)
+	claimLostPVC := s.getPVC(ns, pvc, v1.ClaimLost)
+	stuckPVC := s.getPVC(ns, pvc, "")
+	normalGetFunc :=  func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		return
+	}
+	deadlineExceededGetFunc := func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		return true, nil, pkgerrors.Wrapf(context.DeadlineExceeded, "some wrapped error")
+	}
+	// side effect of poll.Wait. See https://github.com/kubernetes/kubernetes/blob/42fec42586c4411f649bf766fe65059602ee6499/vendor/golang.org/x/time/rate/rate.go#L247
+	preemptiveDeadlineExceededGetFunc := func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		return true, nil, pkgerrors.Wrapf(fmt.Errorf("rate: Wait(n=1) would exceed context deadline"), "some wrapped error")
+	}
```

Comment:
Just feels like the source and test code will be simpler if we just always check for and retrieve the warning events regardless of what the error kinds are.